indexers/common.rb

```diff
@@ -174,6 +174,68 @@ def record_is_umich(r, context)
   acc.compact!
 end
 
+#changes by mrio Feb 2022
+to_field "main_author_display", extract_marc("100abcdefgjklnpqtu4:101abcdefgjklnpqtu4:110abcdefgjklnpqtu4:111abcdefgjklnpqtu4")
+to_field "main_author", extract_marc("100abcdgjkqu:101abcdgjkqu:110abcdgjkqu:111abcdgjkqu")
+
+def concat_subfields(subfields, f)
+  subfields.split('').map{|x| f[x]}.compact.join(" ")
+end
+to_field "contributors_display" do |rec, acc|
```

Comment:
You could probably leverage [`extract_marc_unless`](https://github.com/mlibrary/umich_catalog_indexing/blob/a910dae6eaf10b15e987b6926033959157a24d83/lib/ht_traject/ht_macros.rb#L8-L79) here. It's the same as `extract_marc`, but takes a lambda of the form `my_skip_function(rec, field) -> boolean` -- the two arguments are the whole record and the specific field under consideration. You could use a lambda like `skipif = ->(rec, field) { field.indicator2 == "2" && f["t"].nil? }` and not have to go through all this messing around.

Reply:
You can see [an example of its use](https://github.com/mlibrary/umich_catalog_indexing/blob/ccd6cd170c26ce3a1bd4310ada99ae3d0ca22b31/indexers/common.rb#L147-L157) where I had to ignore JSTOR claiming authorship.

Reply:
And given that you're also messing with the 7xx author fields, you'll want to incorporate the `skipWaSeSS` check as well.

Reply:
Done! I knew there was a better way to do this.